server/src/main/java/org/elasticsearch/persistent/UpdatePersistentTaskStatusAction.java

```diff
@@ -138,7 +138,7 @@ public TransportAction(
         @Override
         protected ClusterBlockException checkBlock(Request request, ClusterState state) {
             // Cluster is not affected but we look up repositories in metadata
-            return state.blocks().globalBlockedException(ClusterBlockLevel.METADATA_WRITE);
+            return state.blocks().globalBlockedException(projectResolver.getProjectId(), ClusterBlockLevel.METADATA_WRITE);
```

Comment:
This is similar to `CompletionPersistentTaskAction` in that persistent tasks can be either project or cluster scoped. I'd leave this to a different PR that focuses on persistent tasks. What do you think?